CRM/Financial/BAO/Payment.php

```diff
@@ -188,6 +188,7 @@ public static function create(array $params): CRM_Financial_DAO_FinancialTrxn {
           'is_post_payment_create' => TRUE,
           'is_email_receipt' => $params['is_send_contribution_notification'],
           'trxn_date' => $params['trxn_date'],
+          'trxn_id' => $params['trxn_id'] ?? NULL,
```

Comment:
This line has caused a regression and change in behaviour in CiviCRM core - see comment on main PR and test failures https://github.com/SemperIT/CiviCARROT/actions/runs/5370451060/jobs/9742545452